indico/modules/events/persons/controllers.py

```diff
@@ -382,6 +385,23 @@ def _process(self):
         return jsonify_form(form)
 
 
+class RHDeleteUnusedEventPerson(RHPersonsBase):
+    def _process_args(self):
+        RHPersonsBase._process_args(self)
+        self.person = EventPerson.query.with_parent(self.event).filter_by(id=request.view_args['person_id']).one()
+
+    def _is_unused(self):
+        persons = self.get_persons()
+        person = persons.get(self.person.email or self.person.id)
+        return not person or person['roles'].get('no_roles', False)
+
+    def _process(self):
+        if self._is_unused():
```

Comment:
I'd rather raise an exception (e.g. Forbidden) when trying to delete a person that's still used